ocfweb/api/session_tracking.py

```diff
@@ -0,0 +1,131 @@
+import json
+from enum import Enum
+from functools import partial
+from ipaddress import ip_address
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.http import HttpResponseBadRequest
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_POST
+from ipware.ip import get_real_ip
+from ocflib.infra.hosts import hosts_by_filter
+from ocflib.infra.net import is_ocf_ip
+from ocflib.lab.stats import get_connection
+
+from ocfweb.caching import cache
+from ocfweb.caching import periodic
+
+CLEANUP_TIMEOUT = 3
+
+State = Enum('State', ['active', 'cleanup'])
+
+get_connection = partial(
+    get_connection,
+    user=settings.OCFSTATS_USER,
+    password=settings.OCFSTATS_PASSWORD,
+    db=settings.OCFSTATS_DB,
+)
+
+
+@require_POST
+@csrf_exempt
+def log_session(request):
+    """Primary API endpoint for session tracking.
+
+    Desktops have a cronjob that calls this endpoint: https://git.io/vpIKX
+    """
+
+    remote_ip = get_real_ip(request)
+
+    if not is_ocf_ip(ip_address(remote_ip)):
+        return HttpResponse('Not Authorized', status=401)
+
+    try:
+        host = _get_desktops().get(remote_ip)
```

Comment:
Wait, how are we dealing with IPv6 addresses?

Reply:
That's a very good question, I don't think they're handled correctly here.